capstone/src/main/webapp/scripts/post.js

```diff
@@ -1,27 +1,80 @@
 function init() {
-  fetchBlobstoreUrlAndShowForm();
   loadPosts();
   getPollOptions();
-
+  fetchBlobstoreUrlAndShowForm();
 }
 
+let postResponse;
+
 function loadPosts() {
-  fetch('/group-post').then(response => response.json()).then((posts) => {
+  fetch('/group-post').then(response => response.json()).then((postsResp) => {
+    postResponse = postsResp;
+    const posts = postResponse["posts"];
     const allPostsList = document.getElementById('posts-container');
     allPostsList.innerHTML = '';
-    for (var i = 0; i < posts.length; i++) {
-      allPostsList.appendChild(createSinglePost(posts[i]));
+    for (let i = 0; i < posts.length; i++) {
+      allPostsList.appendChild(createSinglePost(posts[i], postResponse["likedPosts"]));
     }
+  }).then(() => {
+    addCommentInputListener();
+    addLikeButtonListener();
   });
 }
 
-function createSinglePost(post) {
+function addCommentInputListener() {
+  let elements = document.getElementsByClassName('post-btn align-vertical comment-btn');
+    for (let i = 0; i < elements.length; i++) {
+      elements[i].addEventListener("click", function() {
+        postComment(this.id, this.id + "comment-input")
+      });
+    }
+}
+
+function addLikeButtonListener() {
+  let likeBtns = document.getElementsByClassName('like-icon vertical-align');
+    for (let i = 0; i < likeBtns.length; i++) {
+      likeBtns[i].addEventListener("click", function() {
+        let postId = parseInt(this.id.substring(0, this.id.length - 4));
```

Comment:
not really sure what 4 is but you can store that in a const and since you use 4 in likeToggle as well

Reply:
+1

Reply:
More general comment is that this 4 seems fairly arbitrary and undefined to me. Would a helper function help here? a well named constant? See if you can explain what this 4 means to me in the code

Reply:
I created a function that gets the postId from the likeId since I do this twice. Couldn't think of a constant name that would make sense. If I should clarify anything else, lmk!